restart.go

```diff
@@ -0,0 +1,32 @@
+package selfupdate
+
+import (
+	"os"
+	"syscall"
+
+	"github.com/fynelabs/selfupdate/internal/osext"
+)
+
+func Restart() error {
```

Comment:
I saw the comment about public API, but should this not just be called automatically after Apply in the current model?